internal/storage/image.go

```diff
@@ -966,21 +972,10 @@ func (svc *imageService) CandidatesForPotentiallyShortImageName(systemContext *t
 
 	images := make([]RegistryImageReference, len(resolved.PullCandidates))
 	for i := range resolved.PullCandidates {
-		// Strip the tag from ambiguous image references that have a
-		// digest as well (e.g.  `image:tag@sha256:123...`).  Such
-		// image references are supported by docker but, due to their
-		// ambiguity, explicitly not by containers/image.
```

Comment:
(Absolutely non-blocking: Consider preserving some form of this comment, to have a documentation for the reason we modify the inputs.)